### PR TITLE
Cpp20 compatibility

### PR DESCRIPTION
**Pull request by ThosRTanner**
These changes will make the plugin build with c++20.

Also a couple of small tweaks to remove some unnecessary casting, and a couple of (void) casts to explicitly ignore return values.

### CODE REVIEW

File: stdafx.h [hunk 1]
```diff
@@ -12,10 +12,12 @@
 #include <windows.h>
 
 #define _ATL_CSTRING_EXPLICIT_CONSTRUCTORS    // some CString constructors will be explicit
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
 
 #include <atlbase.h>
 #include <atlstr.h>
 #include <ATLCom.h>
+#pragma comment(lib, "comsuppw.lib")
```
Comment (deadem) [inline stdafx.h]: What is the purpose of the comsuppw.lib here?

Comment (ThosRTanner) [reply stdafx.h]: I get the following link errors:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2001	unresolved external symbol "void __cdecl _com_issue_error(long)" (?_com_issue_error@@YAXJ@Z)	linter	C:\...\notepad-pp-linter\plugin.obj	1	
Error	LNK2001	unresolved external symbol "wchar_t * __cdecl _com_util::ConvertStringToBSTR(char const *)" (?ConvertStringToBSTR@_com_util@@YAPEA_WPEBD@Z)	linter	C:\...\XmlParser.obj	1	
```
without it.

Actually, looking at the error, I just managed to get rid of the 2nd by passing an L-string to bstr_t in one place, just checking on the other one

Comment (ThosRTanner) [reply stdafx.h]: which doesn't appear to be caused by my change

Comment (ThosRTanner) [reply stdafx.h]: However, replacing comutil.h with comdef.h seems to fix it.

Comment (ThosRTanner) [reply stdafx.h]: Updated PR per above.

**Pull request merged**